### PR TITLE
[Modules] Updated Web-Sites API Version & extended common test

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Description

- Updated Web-Sites API Version 
- Extended common test

| Pipeline |
| - |
| [Web: Sites](https://github.com/Azure/ResourceModules/actions/workflows/ms.web.sites.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Web/sites/deploy.bicep [hunk 1]
```diff
@@ -144,6 +144,54 @@ param diagnosticMetricsToEnable array = [
...
+])
+param clientCertMode string = 'Optional'
+
+@description('Optional. If specified during app creation, the app is cloned from a source app.')
+param cloningInfo object = {}
+
+@description('Optional. Size of the function container.')
+param containerSize int = -1
+
+@description('Optional. Unique identifier that verifies the custom domains assigned to the app. Customer will add this id to a txt record for verification.')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Web/sites/deploy.bicep:170]:
```suggestion
@description('Optional. Unique identifier that verifies the custom domains assigned to the app. Customer will add this ID to a txt record for verification.')
```

File: modules/Microsoft.Web/sites/slots/deploy.bicep [hunk 2]
```diff
@@ -142,6 +142,71 @@ param diagnosticMetricsToEnable array = [
...
+])
+param clientCertMode string = 'Optional'
+
+@description('Optional. If specified during app creation, the app is cloned from a source app.')
+param cloningInfo object = {}
+
+@description('Optional. Size of the function container.')
+param containerSize int = -1
+
+@description('Optional. Unique identifier that verifies the custom domains assigned to the app. Customer will add this id to a txt record for verification.')
```
Comment (ahmadabdalla) [inline modules/Microsoft.Web/sites/slots/deploy.bicep:168]:
```suggestion
@description('Optional. Unique identifier that verifies the custom domains assigned to the app. Customer will add this ID to a txt record for verification.')
```

**Review by ahmadabdalla — COMMENTED** [review]
Only two suggestions for the linter test

File: modules/Microsoft.Web/sites/config-appsettings/deploy.bicep [hunk 3]
```diff
@@ -48,7 +48,7 @@ var expandedAppSettings = union(appSettingsKeyValuePairs, azureWebJobsValues, ap
 // =========== //
 // Existing resources //
 // =========== //
-resource app 'Microsoft.Web/sites@2020-12-01' existing = {
+resource app 'Microsoft.Web/sites@2022-03-01' existing = {
```
Comment (eriqua) [inline modules/Microsoft.Web/sites/config-appsettings/deploy.bicep:51]: Please make sure to align with @krbar as he previously opened a PR, currently in draft, which is related #2501 and also has the related issue assigned

Comment (AlexanderSehr) [reply modules/Microsoft.Web/sites/config-appsettings/deploy.bicep:51]: Will do, will ask him to close him PR.

Comment (AlexanderSehr) [reply modules/Microsoft.Web/sites/config-appsettings/deploy.bicep:51]: @eriqua, the PR was closed.

File: modules/Microsoft.Web/sites/deploy.bicep [hunk 4]
```diff
@@ -218,6 +266,17 @@ resource app 'Microsoft.Web/sites@2021-03-01' = {
     keyVaultReferenceIdentity: !empty(keyVaultAccessIdentityResourceId) ? keyVaultAccessIdentityResourceId : null
     virtualNetworkSubnetId: !empty(virtualNetworkSubnetId) ? virtualNetworkSubnetId : any(null)
     siteConfig: siteConfig
+    clientCertEnabled: clientCertEnabled
```
Comment (eriqua) [inline modules/Microsoft.Web/sites/deploy.bicep:269]: I see there a lot of common parameters for parent and child slots. Would it make sense to research if there are any constraints in their values? For instance, is it possible to have clientCertEnabled disabled for the parent and enabled for the slot? I'm not raising this to block the PR, but to double check if we should create a separate issue on that.

Comment (AlexanderSehr) [reply modules/Microsoft.Web/sites/deploy.bicep:269]: As I understood the previous PRs, they are pretty much the same.

Comment (AlexanderSehr) [reply modules/Microsoft.Web/sites/deploy.bicep:269]: To your question - no idea unfortunately if each slot is considered an independent app (which I think is the ideal) I would expect it works

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**